Eastern-Research-Group/mywaterway
# Feature/579 improving download legend

## Related Issues:
* [HMW-579](https://jira.epa.gov/browse/HMW-579)

## Main Changes:
* Updated the legend widget to output a custom legend, since the OOB esri one doesn't show all legend items. One benefit of the custom legend is that it matches the HMW legend more closely.
  * This works by using the OOB esri widget to generate the map portion and then it tacks on the legend.
  * I started off using `jsPDF` and `jsPDF-autotable` to do this, but I found that I needed a 3rd library (`pdf-lib`) to merge two PDF files together. I just ended up using `pdf-lib` for all of the pdf generating logic.
  * I ended up removing the File Format and DPI settings.
* Added labels to unique value renderer. This was originally done because the OOB esri widget needed that to display in the legend. I decided to leave that in there since it adds a little more information to the code and may help us with esri stuff in the feature.
* Changed `All Waterbodies...` titles to `Surrounding Waterbodies...` for consistency.
* Updated the WSIO gradient icon to include the text in the SVG, instead of in a separate div. This makes it easier to output the gradient icon into the PDF.
* Fixed issue of loading spinner not showing up when uploading a file to the add data widget.

## Steps To Test:
1. Navigate to http://localhost:3000/community/020700100201/overview
2. Zoom in one level, so the `Mapped Water (all)` layer is available
3. Turn on all of the layers
4. Use the download widget to save a pdf
5. Verify everything is shown in the PDF
6. Use the Add Data Widget to upload a layer as a file
7. Verify a loading spinner is displayed
8. Expand the legend
9. Verify the `State Watershed Health Index` gradient looks good
10. Go to the Protect tab and expand `Watershed Health Scores`
11. Verify the gradient icon looks good here as well

## TODO:
- [x] Would like to look into making the legend display in multiple columns (2 or 3 depending on chosen layout). I might still try to include that in this PR, but if it doesn't make it that's fine.
